Exception is swallowed

**Describe the bug**
When building with maven (3.9.5, mvn clean install), all that is output is:

```
[INFO] --- dependency-check:9.0.5:aggregate (default) @ my-project ---
[ERROR] Fatal exception(s) analyzing My Project
[INFO] 
[INFO] --- failsafe:2.22.0:verify (default) @ my-project ---
```
To see the problem I ran `mvn clean install -X`.

This is the main cause:

`caused by MVStoreException: The write format 3 is larger than the supported format 2 [2.1.214/5]`

Main cause is obvious, I need to purge / delete the data.

But exception is not logged in maven output (without -X), probably because of: see attached stacktrace
[stacktrace.txt](https://github.com/jeremylong/DependencyCheck/files/13673724/stacktrace.txt)


**Version of dependency-check used**
The problem occurs using version 9.0.5 of the maven plugin

**Log file**
See above.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
An exception should be logged when building with mvn clean install.

**Additional context**
Add any other context about the problem here.

--- a/core/src/main/java/org/owasp/dependencycheck/data/nvdcve/DatabaseManager.java
+++ b/core/src/main/java/org/owasp/dependencycheck/data/nvdcve/DatabaseManager.java
@@ -197,6 +197,9 @@ public final class DatabaseManager {
                         LOGGER.debug("Unable to connect to the database", ex);
                         throw new DatabaseException("Unable to connect to the database", ex);
                     }
+                } else if (isH2 && ex.getMessage().contains("file version or invalid file header")) {
+                    LOGGER.error("Incompatible or corrupt database found. To resolve this issue please remove the existing database by running purge");
+                    throw new DatabaseException("Incompatible or corrupt database found; run the purge command to resolve the issue");
                 } else {
                     LOGGER.debug("Unable to connect to the database", ex);
                     throw new DatabaseException("Unable to connect to the database", ex);

